tests/test_swagger.py

```diff
@@ -170,6 +154,72 @@ def view_put(self, request):
                           ['application/json'])
 
 
+class TestExtractTags(unittest.TestCase):
+
+    def test_user_defined_tags(self):
+
+        service = Service("IceCream", "/icecream/{flavour}")
+
+        class IceCream(object):
+            @service.get(tags=['cold', 'foo'])
+            def view_get(self, request):
+                return service
+
+        swagger = CorniceSwagger([service])
+        spec = swagger('IceCreamAPI', '4.2')
+        validate(spec)
+        tags = spec['paths']['/icecream/{flavour}']['get']['tags']
+        self.assertEquals(tags, ['cold', 'foo'])
+
+    def test_listed_default_tags(self):
+
+        service = Service("IceCream", "/icecream/{flavour}")
+
+        class IceCream(object):
+            @service.get()
+            def view_get(self, request):
+                return service
+
+        swagger = CorniceSwagger([service])
+        spec = swagger('IceCreamAPI', '4.2',
+                       default_tags=['cold'])
+        validate(spec)
+        tags = spec['paths']['/icecream/{flavour}']['get']['tags']
+        self.assertEquals(tags, ['cold'])
+
+    def test_callable_default_tags(self):
+
+        service = Service("IceCream", "/icecream/{flavour}")
+
+        class IceCream(object):
+            @service.get()
+            def view_get(self, request):
+                return service
+
+        def default_tag_callable(service):
+            return ['cold']
+
+        swagger = CorniceSwagger([service])
+        spec = swagger('IceCreamAPI', '4.2',
+                       default_tags=default_tag_callable)
+        validate(spec)
+        tags = spec['paths']['/icecream/{flavour}']['get']['tags']
+        self.assertEquals(tags, ['cold'])
+
+    def test_invalid_tag_raises_exception(self):
+
+        service = Service("IceCream", "/icecream/{flavour}")
+
+        class IceCream(object):
+            @service.get(tags='cold')
+            def view_get(self, request):
+                return service
+
+        swagger = CorniceSwagger([service])
+        self.assertRaises(CorniceSwaggerException,
+                          swagger, 'IceCreamAPI', '4.2')
```

Comment:
It's tested on `test_generate_swagger_spec`. The tests from version 0.2.1 tests that are still supported.